components/signup/signup_email/index.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {ActionCreatorsMapObject, bindActionCreators, Dispatch} from 'redux';
+
+import {GenericAction, ActionFunc} from 'mattermost-redux/types/actions';
+import {UserProfile} from 'mattermost-redux/types/users';
+import {ServerError} from 'mattermost-redux/types/errors';
+
+import {createUser} from 'mattermost-redux/actions/users';
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
+import {getTeamInviteInfo} from 'mattermost-redux/actions/teams';
+
+import {setGlobalItem} from 'actions/storage';
+import {loginById} from 'actions/views/login';
+import {getPasswordConfig} from 'utils/utils.jsx';
+
+import {GlobalState} from '../../../types/store';
+
+import SignupEmail from './signup_email';
+
+export type PasswordConfig = {
+    minimumLength: number;
+    requireLowercase: boolean;
+    requireNumber: boolean;
+    requireSymbol: boolean;
+    requireUppercase: boolean;
+}
+
+export type Props = {
```

Comment:
Can you move these definitions into `signup_email.tsx` and re-export them from there if necessary? That's generally where we define all of the Props and State types unless the `mapStateToProps` needs additional props of its own

Reply:
The reason I did this was because I wasn't sure if exporting it from signup_email.tsx would cause a circular import issue or not due to needing Props and State for the connect workaround. I'll move them back where they belong if they don't cause any issues.

Reply:
I'd think having them in `signup_email.tsx` should prevent circular references since `index.ts` would import `signup_email.tsx` and that wouldn't happen the other way around. That said, if you do run into that, using `import types` to just import the types should prevent circular references causing any problems as long as its just with type defintions.